src/middleware/authMiddleware.js

```diff
@@ -20,14 +20,8 @@ export const hsesAuth = new ClientOAuth2({
  * @param {*} res - response
  */
 export function login(req, res) {
-  if (process.env.NODE_ENV !== 'production'
-    && req.headers.cookie && req.headers.cookie === `CUCUMBER_USER=${process.env.CUCUMBER_USER}`) {
-    req.session.userId = process.env.CUCUMBER_USER_ID;
-    res.redirect(process.env.TTA_SMART_HUB_URI);
-  } else {
-    const uri = hsesAuth.code.getUri();
-    res.redirect(uri);
-  }
+  const uri = hsesAuth.code.getUri();
+  res.redirect(uri);
```

Comment:
cucumber should be using only the `authMiddleware` function

Reply:
Tell me more about it. Was there a specific reason the check was moved to authMiddleware?

Reply:
There's no way for the accessibility scanner to select the "login" button on requests to the root page because it's not a web crawler. So the `login` function isn't used for accessibility scans. But we still hit use the `authMiddleware` on all requests to root.  So this check and userId set was moved to that function. The cucumber script looks like it only attempts to select the login button if not logged in (https://github.com/adhocteam/Head-Start-TTADP/blob/fc68f174e623ba383e662be77737cfeb85b2f919/cucumber/features/steps/homePageSteps.js#L47) so started at a "logged in state" shouldn't harm it.

Reply:
I see, so there was no way to set the session variable for the accessibility scanner. Is that right?

If we are moving the cucumber code, there are a couple of items to also add/change.
1.  Remove the session code from cucumber.
2.  Modify the `.env.example` with with the variable changes.
3.  Remove the env var CUCUMBER_USER (not sure if that was already done in CircleCI).
4.  Add a unit test for the added code so we are not losing code coverage.

Reply:
@kryswisnaskas Thanks for this comment. I added both a happy path and a sad path test for the `authMiddleware` function. I did **not** remove the `CUCUMBER_USER_ID` related code. I think we're likely to need cucumber tests that require user attributes in the very near future. Am I off base here?

Reply:
The tests look good. There is only one problem that couple of other tests fail locally. This is because in a non-prod environment, the condition` if (process.env.NODE_ENV !== 'production' && process.env.BYPASS_AUTH === 'true')` will always be true and we will never hit the `res.sendStatus(401)` line that we are also testing.

Reply:
I think it makes sense to remove the unneeded session cookies setting code that with this change we are no longer using. This way we are not creating any tech debt, but can always go back to the previous code if useful. 
All references to `CUCUMBER_USER` (but not `CUCUMBER_USER_ID`) can now be removed (package.json, .example.env and homePageSteps.js). 

In `homePageSteps.js`, the lines 23-36 set the page cookie.

In `.example.env` the line referencing `CUCUMBER_USER` can be substituted with
`BYPASS_AUTH=true`

Reply:
@kryswisnaskas I think I've addressed all your concerns. Thanks for your help on this.